internal/provider/provider.go

```diff
@@ -448,13 +451,23 @@ func (p *MongodbtlasProvider) Resources(context.Context) []func() resource.Resou
 	return resources
 }
 
-func NewFrameworkProvider() provider.Provider {
-	return &MongodbtlasProvider{}
+func NewFrameworkProvider(proxyPort *int) provider.Provider {
+	return &MongodbtlasProvider{
+		proxyPort: proxyPort,
+	}
 }
 
 func MuxedProviderFactory() func() tfprotov6.ProviderServer {
-	v2Provider := NewSdkV2Provider()
-	newProvider := NewFrameworkProvider()
+	return muxedProviderFactory(nil)
+}
+
+func MuxedProviderFactoryForTesting(proxyPort *int) func() tfprotov6.ProviderServer {
```

Comment:
knitpit maybe just call mux instead of muxed, e.g. MuxProviderFactoryForTesting,  muProviderFactory, etc.
(HC uses tf6muxserver.NewMuxServer, etc.)